subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java

```diff
@@ -108,4 +112,32 @@ public String transform(GradleException e) {
             }
         }), LINE_SEPARATOR + "AND" + LINE_SEPARATOR);
     }
+
+    private class ParentBuildOperationAwareWorker<T extends BuildOperation> implements BuildOperationWorker<T> {
+        private final BuildOperationExecutor.Operation parentOperation;
+        private final BuildOperationWorker<T> delegate;
+
+        private ParentBuildOperationAwareWorker(BuildOperationWorker<T> delegate) {
+            this.parentOperation = buildOperationExecutor.getCurrentOperation();
+            this.delegate = delegate;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return delegate.getDisplayName();
+        }
+
+        @Override
+        public void execute(final T t) {
+            BuildOperationDetails operationDetails = BuildOperationDetails.displayName(t.getDescription())
```

Comment:
For the record and future reviewers, we decided to defer providing more information here to a subsequent PR. Basically, we could have a `DescribableBuildOperation` interface which gives us more than just the description: display name, operation metadata, ...